diff --git a/hyprland-share-picker/main.cpp b/hyprland-share-picker/main.cpp
index d09a27d25ae..f89434e99c4 100644
--- a/hyprland-share-picker/main.cpp
+++ b/hyprland-share-picker/main.cpp
@@ -111,5 +111,5 @@ int main(int argc, char* argv[]) {
         QString      text = QString::fromStdString(std::string("Screen " + std::to_string(i) + " at " + std::to_string(GEOMETRY.x()) + ", " + std::to_string(GEOMETRY.y()) + " (" +
                                                           std::to_string(GEOMETRY.width()) + "x" + std::to_string(GEOMETRY.height()) + ") (") +
-                                              SCREENS[i]->name().toStdString() + ")");
+                                              SCREENS[i]->name().toUtf8().constData() + ")");
         QString outputName = SCREENS[i]->name();
         ElidedButton* button = new ElidedButton(text);
@@ -122,5 +122,5 @@ int main(int argc, char* argv[]) {
             std::cout << "/";

-            std::cout << "screen:" << outputName.toStdString() << "\n";
+            std::cout << "screen:" << outputName.toUtf8().constData() << "\n";

             settings->setValue("width", mainPickerPtr->width());
@@ -179,5 +179,5 @@ int main(int argc, char* argv[]) {

     QString      text = "Select region...";
-
+
     ElidedButton* button = new ElidedButton(text);
     button->setMaximumSize(400, BUTTON_HEIGHT);
@@ -199,5 +199,5 @@ int main(int argc, char* argv[]) {

         for (auto& screen : SCREENS) {
-            if (screen->name().toStdString() == SCREEN_NAME) {
+            if (screen->name().toUtf8().constData() == SCREEN_NAME) {
                 pScreen = screen;
                 break;
