diff --git a/src/debug/Log.cpp.old b/src/debug/Log.cpp
index a4c5b08e60b..0def77c003e 100644
--- a/src/debug/Log.cpp.old
+++ b/src/debug/Log.cpp
@@ -6,11 +6,8 @@
 #include <fstream>
 #include <iostream>
-#include <fcntl.h>

 void Debug::init(const std::string& IS) {
     logFile = IS + (ISDEBUG ? "/hyprlandd.log" : "/hyprland.log");
     logOfs.open(logFile, std::ios::out | std::ios::app);
-    auto handle = logOfs.native_handle();
-    fcntl(handle, F_SETFD, FD_CLOEXEC);
 }

